xml-parser/src/main/java/net/adoptopenjdk/icedteaweb/xmlparser/XMLSanitizer.java

```diff
@@ -68,7 +68,8 @@ static String sanitizeXml(final Reader in) {
                         charInBuffer--;
                     }
                 } else {
-                    if (buffer[0] == '<' && buffer[1] == '!' && buffer[2] == '-' && buffer[3] == '-') {
+                    // jnlp files from a MFSys25  contain '<?-- JViewerVersion 3.30a -->' therefore we must treat ! and ? equally
+                    if (buffer[0] == '<' && (buffer[1] == '!' || buffer[1] == '?') && buffer[2] == '-' && buffer[3] == '-') {
```

Comment:
@sclassen can we add a unit test?

Reply:
Maybe next year stuck_out_tongue_closed_eyes